core/chains/evm/log/broadcaster.go

```diff
@@ -204,21 +214,34 @@ func (b *broadcaster) awaitInitialSubscribers() {
 }
 
 func (b *broadcaster) Register(listener Listener, opts ListenerOpts) (unsubscribe func()) {
-	if len(opts.LogsWithTopics) == 0 {
-		b.logger.Panic("Must supply at least 1 LogsWithTopics element to Register")
-	}
+	// IfNotStopped RLocks the state mutex so LB cannot be closed until this
+	// returns (no need to worry about listening for b.chStop)
+	//
+	// NOTE: We do not use IfStarted here because it is explicitly ok to
+	// register listeners before starting, this allows us to register many
+	// listeners then subscribe once on start, avoiding thrashing
+	ok := b.IfNotStopped(func() {
+		if len(opts.LogsWithTopics) == 0 {
+			b.logger.Panic("Must supply at least 1 LogsWithTopics element to Register")
+		}
 
-	reg := registration{listener, opts}
-	wasOverCapacity := b.addSubscriber.Deliver(reg)
-	if wasOverCapacity {
-		b.logger.Error("Subscription mailbox is over capacity - dropped the oldest unprocessed subscription")
-	}
-	return func() {
-		wasOverCapacity := b.rmSubscriber.Deliver(reg)
+		sub := &subscriber{listener, opts}
+		wasOverCapacity := b.changeSubscriberStatus.Deliver(changeSubscriberStatus{subscriberStatusSubscribe, sub})
 		if wasOverCapacity {
-			b.logger.Error("Subscription removal mailbox is over capacity - dropped the oldest unprocessed removal")
+			panic("LogBroadcaster subscribe: changeSubscriberStatus channel was full")
```

Comment:
For what reason you used `panic` instead of `b.logger.Panic`?

Reply:
No reason, I will fix.